File: codegen/results/v001/qwen3.5_27b_think/20260408_105307/go/002_word_frequency/solution.go
package main

import (
	"fmt"
	"os"
	"sort"
	"strings"
	"unicode"
)

type WordCount struct {
	Word  string
	Count int
}

func main() {
	// Read the file
	content, err := os.ReadFile("input/text.txt")
	if err != nil {
		panic(err)
	}

	// Convert to lowercase
	text := strings.ToLower(string(content))

	// Strip punctuation: keep only letters, replace others with space
	var cleaned strings.Builder
	cleaned.Grow(len(text))
	for _, r := range text {
		if unicode.IsLetter(r) {
			cleaned.WriteRune(r)
		} else {
			cleaned.WriteRune(' ')
		}
	}

	// Split into words based on whitespace
	words := strings.Fields(cleaned.String())

	// Count frequencies
	frequencies := make(map[string]int)
	for _, word := range words {
		frequencies[word]++
	}

	// Convert map to slice for sorting
	var wordCounts []WordCount
	for word, count := range frequencies {
		wordCounts = append(wordCounts, WordCount{Word: word, Count: count})
	}

	// Sort by count descending, then word alphabetically ascending
	sort.Slice(wordCounts, func(i, j int) bool {
		if wordCounts[i].Count != wordCounts[j].Count {
			return wordCounts[i].Count > wordCounts[j].Count
		}
		return wordCounts[i].Word < wordCounts[j].Word
	})

	// Output results
	for _, wc := range wordCounts {
		fmt.Printf("%s: %d\n", wc.Word, wc.Count)
	}
}

File: codegen/results/v001/qwen3.5_27b_think/20260408_105307/typescript/002_word_frequency/solution.ts
import * as fs from 'fs';
import * as path from 'path';

const filePath = path.resolve('input/text.txt');
const content = fs.readFileSync(filePath, 'utf-8');

const normalized = content
  .toLowerCase()
  .replace(/[^a-z\s]/g, '');

const words = normalized.split(/\s+/).filter(word => word.length > 0);

const frequency: Map<string, number> = new Map();
for (const word of words) {
  frequency.set(word, (frequency.get(word) || 0) + 1);
}

const sorted = Array.from(frequency.entries())
  .sort((a, b) => {
    if (b[1] !== a[1]) {
      return b[1] - a[1];
    }
    return a[0].localeCompare(b[0]);
  });

for (const [word, count] of sorted) {
  console.log(`${word}: ${count}`);
}